File: src/com/codingsimply/apps/WakeOnLan.java
package com.codingsimply.apps;

import com.codingsimply.apps.core.data.Computer;
import com.codingsimply.apps.utility.WakeUp;

import java.io.IOException;

public class WakeOnLan {

    private static final int PORT = 9;

    public static void main(String[] args) {
        Computer computer = new Computer("192.168.1.2", "00:00:00:00:00:00");
        try {
            WakeUp wakeUp = new WakeUp(PORT);
            wakeUp.now(computer);
            System.out.println("Wake-on-LAN packet sent.");
        } catch (IOException | IllegalArgumentException e) {
            System.out.println("Failed to send Wake-on-LAN packet: + e");
            System.exit(1);
        }
    }
}

File: src/com/codingsimply/apps/WakeOnLan.kt
package com.codingsimply.apps

import com.codingsimply.apps.core.data.Computer
import com.codingsimply.apps.utility.WakeUp

import java.io.IOException

object WakeOnLan {

    private val PORT = 9

    @JvmStatic
    fun main(args: Array<String>) {
        val ip = args[0]
        val mac = args[1]

        val computer = Computer(ip, mac)
        try {
            val wakeUp = WakeUp(PORT)
            wakeUp.now(computer)
            println("Wake-on-LAN packet sent.")
        } catch (e: IOException) {
            println("Failed to send Wake-on-LAN packet: + e")
            System.exit(1)
        } catch (e: IllegalArgumentException) {
            println("Failed to send Wake-on-LAN packet: + e")
            System.exit(1)
        }

    }
}
